rosetta/services/tx_operation.go

```diff
@@ -127,15 +134,94 @@ func GetNativeOperationsFromStakingTransaction(
 		Account:  accountID,
 		Amount:   amount,
 		Metadata: metadata,
-	}), nil
+	})
+
+	// expose delegated balance
+	if tx.StakingType() == stakingTypes.DirectiveDelegate {
+		op2 := GetDelegateOperationForSubAccount(tx, operations[1])
+		return append(operations, op2), nil
+	}
+
+	if tx.StakingType() == stakingTypes.DirectiveUndelegate {
+		// set sub account
+		validatorAddress := operations[1].Metadata["validatorAddress"]
```

Comment:
I love the refactor you did in `GetDelegateOperationForSubAccount`. Can we also do it for undelegate?

Reply:
Yep.